File: wennv/wenlv/src/api/checkin.ts
/** 打卡接口封装（数据永久落库 MySQL check_ins 表，护照集章凭证） */
import { get, post, del } from './request'

export interface CheckInItem {
  id: number
  scenic_id: number
  photo_url?: string
  /** 打卡照片 URL 列表（后端由 JSON 列解析为数组） */
  photos: string[]
  comment?: string
  visited_at: string
  created_at: string
}

/** 打卡（photos 必传至少 1 张，均为 OSS 直传后的公开 URL） */
export function createCheckIn(payload: {
  scenic_id: number
  photos: string[]
  comment?: string
  visited_at?: string
}) {
  return post<void>('/check-ins', payload)
}

/** 我的打卡列表 */
export function listCheckIns(): Promise<CheckInItem[]> {
  return get<CheckInItem[]>('/check-ins')
}

/** 删除打卡 */
export function removeCheckIn(id: number) {
  return del<void>(`/check-ins/${id}`)
}

File: houduan/model/checkin.go
package model

import "time"

// CheckIn 打卡记录,可携带用户上传的游玩照片 URL 与评论。
// Photos 存 JSON 数组(最多 9 张,前端浏览器直传 OSS 后回传 URL 列表);
// PhotoURL 保留为首图(与旧数据/旧接口兼容)。
type CheckIn struct {
	ID        uint      `gorm:"primaryKey;comment:打卡ID" json:"id"`
	UserID    uint      `gorm:"index:idx_uid_scenic,unique;comment:用户ID" json:"user_id"`
	ScenicID  uint      `gorm:"index:idx_uid_scenic,unique;comment:景点ID" json:"scenic_id"`
	PhotoURL  string    `gorm:"size:512;comment:打卡首图URL(兼容旧字段)" json:"photo_url"`
	Photos    string    `gorm:"size:4096;comment:打卡照片URL列表(JSON数组,最多9张)" json:"photos"`
	Comment   string    `gorm:"size:255;comment:打卡感言" json:"comment"`
	VisitedAt time.Time `gorm:"comment:到访时间" json:"visited_at"`
	CreatedAt time.Time `gorm:"comment:创建时间" json:"created_at"`
}
